### DISCUSSION TITLE
v1.1-20220404

### DISCUSSION THREAD

**Discussion by erichelgeson** [erichelgeson/BlueSCSI-v1#87] [General]
# Changes

* Apple HD SC Setup support #71
  * You can now format in most Mac utilities without error.
* Image set selection #73 
  * Add a physical switch/jumpers externally to your BlueSCSI so you can switch computers without moving files on the SD card. Select between 4 image sets (folders of images) on the SD card. There is also an optional shared image set.
  * See Github issue for usage instructions.
* CLZ optimization #63, Optimize handling of reset interrupt. #64, Code reordering for bus settle delays #78, Upgrade to latest toolchain #67, #70
  * These changes give a performance improvement of 10%+ and better stability. My 6100 does 1.2mb/sec read, 1.2mb/sec write.
* Be more lenient on SD cards that are not able to communicate at 50MHz SPI #69
  * Some SD Cards would fail to communicate with the BlueSCSI at 50MHz but work fine at 49mhz. No real change in performance till you reach 40MHz, which a warning will be logged.
* Fixes to get VAXstation 3100 working #59

Compatible with Hardware versions 1.0-a, b, c and 1.1-a.

# Contributors

@mactcp, @pacjunk, @erichelgeson,  @marcelv-3, moof (Discord)

# Flashing

To update you will need a STLink V2 clone programmer and follow the flashing instructions here https://github.com/erichelgeson/BlueSCSI/wiki/Flashing#using-stlink-v2-preferred

Note USB flashing is currently not possible, though we hope to add it in a future release.

**Full Changelog**: https://github.com/erichelgeson/BlueSCSI/compare/v1.1-20211215...v1.1-20220404

<hr /><em>This discussion was created from the release <a href='https://github.com/erichelgeson/BlueSCSI/releases/tag/v1.1-20220404'>v1.1-20220404</a>.</em>

**Comment by erichelgeson** [comment]
Thank you to everyone who contributed and tested this release! Please note docs will be updated through the week as we have time.

**Comment by MrGasS** [comment]
I've installed the ImageSet switches on my homemade BlueSCSI board and damn, this function is God's hand.
Now I don't need to deal with multiple SD card anymore.

Thank you for everything.

![IMG_20220406_195650234_](https://user-images.githubusercontent.com/37547183/162038873-e840c959-4dc5-4335-bf34-06ea2035ca02.jpg)

**Reply by erichelgeson** [reply]
Wow, this is awesome! I love seeing custom boards! You're probably only the 3rd person in the world to have the image switcher in hardware  - and it looks good :)

**Reply by MrGasS** [reply]
Thank you a lot! :)
While the blue board is newer and I made it the last saturday for external use, tonight I installed the switches into my first BlueSCSI that I made in February.
In fact, this BlueSCSI replaced a 1 GB Seagate hard disk into my main 68k machine, a Quadra 700, which contains a lot of personal and sensitive data.
However, while a 4 GB hardfile with 7.6.1 is always on, the first combination enables a 8.1 disk, the second combination enables an OS restore disk, the last one enables an hardfile with my private data.
I'm loving this feature.

![IMG_20220407_211128736](https://user-images.githubusercontent.com/37547183/162279752-3a5b429e-97b7-4f32-9954-4c830c3c84c5.jpg)

**Comment by erichelgeson** [comment]
USB flashing is now possible with the BlueSCSI Updater - Though you'll need the USB boot loader first which does require a STLink. The USB bootlaoder is now shipping on all BlueSCSI's going forward. 

Read the readme for full details here:
https://github.com/erichelgeson/BlueSCSI-Updater/

**Comment by axelmuhr** [comment]
"_You're probably only the 3rd person in the world to have the image switcher in hardware_"
4th person here ;-)

That said, I have issues with newer BluePills which Identify as 0x414/Rev Z but report 256KB flash(??).
Using these, booting ends in MacsBug close to entering the Finder.

My older BluePills work fine on this, though.

![20220525_101444](https://user-images.githubusercontent.com/5321172/170215231-c1e1cba5-fb82-405b-b64b-72feafc8bc35.jpg)

**Reply by erichelgeson** [reply]
The 256kb chips are likely APM32 rebadged. You can get them working by setting the clock speed to 96MHz - I have a profile in the main branch now for that. I'll also be doing a release soon so you can just flash a bin for it too. https://github.com/erichelgeson/BlueSCSI/pull/110

Nice looking board!

**Reply by axelmuhr** [reply]
Thanks! Yeah, it's an all-in-one (for internal mounting _and_ external use), offering 3 ways of powering it externally (5.25" and 3.5" floppy as well as USB) and connectors for a break-out sd-card reader... the config permutations are quite big ;-)

Re: APM32 - I tried many clockings (30, 48, 60, 96 and even 128MHz).
-- deleted my previous babble -- stupid me tweaked the SPI clock 😒... 
As I'm using ST-Link I need to adjust your STM32F1-USB-96MHz a bit first.
**Update**: 
Just saw in [this table](https://mecrisp-stellaris-folkdoc.sourceforge.io/bluepill-diagnostics-v1.6.html#table-device-info), that ID 0x414 isn't an APM32 but rather an STM32F103VET6 or GD32F103VE - sadly I'm unable to find some Q'a'D identification tool (like blinky.c which seems to be vanished).

**Reply by axelmuhr** [reply]
Ok, 96MHz did the trick with this, too 👍
